Remove unnecessary update API call for client-side-only field changes in google_bigquery_table

wj-chen opened the pull request:
<!--
Complete the self-review checklist to help speed up the review process: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

If your PR is still work in progress, please create it in draft mode.

Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to.
For example: Fixes https://github.com/hashicorp/terraform-provider-google/issues/ISSUE_ID
-->
Fixes https://github.com/hashicorp/terraform-provider-google/issues/19485, following instructions added in https://github.com/GoogleCloudPlatform/magic-modules/pull/11330.

Tested locally with `TF_LOG=DEBUG terraform apply` for client-side-only field changes and mixed changes.

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:none
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12174-old..auto-pr-12174) ( 1 file changed, 13 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12174-old..auto-pr-12174) ( 1 file changed, 13 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: 121
Passed tests: 110
Skipped tests: 10
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>bigquery</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccBigQueryTable_allowDestroy</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccBigQueryTable_allowDestroy` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/d22aa466-a869-4992-95b3-d2c6510ce9e9/recording/TestAccBigQueryTable_allowDestroy.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---






🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/d22aa466-a869-4992-95b3-d2c6510ce9e9/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/d22aa466-a869-4992-95b3-d2c6510ce9e9/recording) for each test

ScottSuarez reviewed (changes requested):
Can you add an update test for this ?
https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test

Should be relatively simple. Just want to make sure that this works going forward. My only concern is that we don't `return resourceBigQueryTableRead(d, meta)` like the original function does. I see you've tested locally and I don't foresee any problem but it would be great to capture this behavior in a test!

wj-chen commented:
> Can you add an update test for this ? https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test
> 
> Should be relatively simple. Just want to make sure that this works going forward. My only concern is that we don't `return resourceBigQueryTableRead(d, meta)` like the original function does. I see you've tested locally and I don't foresee any problem but it would be great to capture this behavior in a test!

Added a test. And good point about `return resourceBigQueryTableRead(d, meta)`. I added it back since if the table was deleted outside of Terraform before the update, we should detect that correctly and treat it as a creation again. Thanks for the suggestion.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12174-old..auto-pr-12174) ( 2 files changed, 70 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12174-old..auto-pr-12174) ( 2 files changed, 70 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: 123
Passed tests: 111
Skipped tests: 10
Affected tests: 2

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>bigquery</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccBigQueryTable_OnlyDeletionProtectionUpdate</li>
<li>TestAccBigQueryTable_allowDestroy</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccBigQueryTable_OnlyDeletionProtectionUpdate` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/5b33440b-16c0-47fe-8427-d6f902887a58/recording/TestAccBigQueryTable_OnlyDeletionProtectionUpdate.log)]
`TestAccBigQueryTable_allowDestroy` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/5b33440b-16c0-47fe-8427-d6f902887a58/recording/TestAccBigQueryTable_allowDestroy.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---






🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/5b33440b-16c0-47fe-8427-d6f902887a58/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/5b33440b-16c0-47fe-8427-d6f902887a58/recording) for each test

ScottSuarez commented on mmv1/third_party/terraform/services/bigquery/resource_bigquery_table.go:
```diff
@@ -1903,6 +1903,19 @@ type TableReference struct {
+	clientSideFields := map[string]bool{"deletion_protection": true}
+	clientSideOnly := true
+	for field := range ResourceBigQueryTable().Schema {
+		if d.HasChange(field) && !clientSideFields[field] {
```
I assume nested changes will be caught by `d.HasChange(field)`. Would be possible adding a test only changing a nested subfield?

wj-chen replied:
Sure, added.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12174-old..auto-pr-12174) ( 2 files changed, 163 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12174-old..auto-pr-12174) ( 2 files changed, 163 insertions(+), 1 deletion(-))

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12174-old..auto-pr-12174) ( 2 files changed, 163 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12174-old..auto-pr-12174) ( 2 files changed, 163 insertions(+), 1 deletion(-))

modular-magician commented:
#### Tests analytics
Total tests: 126
Passed tests: 115
Skipped tests: 10
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>bigquery</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccBigQueryTable_OnlyNestedFieldUpdate</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccBigQueryTable_OnlyNestedFieldUpdate` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/ffff1d66-ed2e-4283-9e24-453348090176/recording/TestAccBigQueryTable_OnlyNestedFieldUpdate.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---






🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/ffff1d66-ed2e-4283-9e24-453348090176/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/ffff1d66-ed2e-4283-9e24-453348090176/recording) for each test

modular-magician commented:
#### Tests analytics
Total tests: 126
Passed tests: 116
Skipped tests: 10
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>bigquery</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12174/artifacts/34f18f7d-948c-4a4e-ac5c-8f51596dc97d/build-log/replaying_test.log)

ScottSuarez commented:
Thanks for the work!
